### PR TITLE
Added skills, reworked skill creation, create event-trigger system for new skills, added rewards for each skill and tidied up command output

### PR DESCRIPTION
**Pull request by CallumJohnson**
As well as adding skills, skills have also been modified, combined and therefore removed respectively.

We have now got Athlete, Boating, Breeding, Cardio, Crafting, Digging, Dueling, Enchanting, Farming, Fishing, Floriculture, Gliding, Hardiness, LumberJack, Mining, Monster Hunting, Pyromaniac, Quarrying, Riding and Strength.

### CODE REVIEW

File: src/main/java/dansplugins/simpleskills/SimpleSkills.java [hunk 1]
```diff
@@ -156,20 +151,10 @@ private void setTabCompleterForCoreCommands() {
...
-                new EnchantingHandler(),
-                new MoveHandler(),
-                new BreedingHandler(),
-                new DeathHandler()
-        ));
-        EventHandlerRegistry eventHandlerRegistry = new EventHandlerRegistry(getPonderAPI());
-        eventHandlerRegistry.registerEventHandlers(listeners, this);
+    private void registerEvents() {
+        for (AbstractSkill skill : PersistentData.getInstance().getSkills()) {
+            skill.register();
```
Comment (dmccoystephenson) [inline src/main/java/dansplugins/simpleskills/SimpleSkills.java:156]: This is beautiful.

File: src/main/java/dansplugins/simpleskills/commands/SkillCommand.java [hunk 2]
```diff
@@ -1,40 +1,42 @@
...
-        Skill skill = PersistentData.getInstance().getSkill(skillName);
+        AbstractSkill skill = PersistentData.getInstance().getSkill(skillName);
         if (skill == null) {
-            commandSender.sendMessage(LocalMessageService.getInstance().convert(LocalMessageService.getInstance().getlang().getString("SkillNotFound")));
-
+            commandSender.sendMessage(LocalMessageService.getInstance()
+                    .convert(LocalMessageService.getInstance().getlang().getString("SkillNotFound")));
         }
-        skill.sendInfo(commandSender);
+        // TODO fix skill.sendInfo(commandSender);
```
Comment (dmccoystephenson) [inline src/main/java/dansplugins/simpleskills/commands/SkillCommand.java:38]: What needs to be fixed here?

File: src/main/java/dansplugins/simpleskills/skills/Pyromaniac.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+     *
+     * @param player    to reward.
+     * @param skillData assigned data to the skill reward, 'Block' for 'BlockSkills' etc.
+     */
+    @Override
+    public void executeReward(@NotNull Player player, Object... skillData) {
+        final PlayerRecord record = getRecord(player);
+        if (record == null) return;
+        if (!ChanceCalculator.getInstance().roll(record, this, 0.10)) return;
+        player.sendMessage("§bThe fire has made you §6crazy§b!");
```
Comment (dmccoystephenson) [inline src/main/java/dansplugins/simpleskills/skills/Pyromaniac.java:104]: I like this lol

File: src/main/java/dansplugins/simpleskills/skills/Riding.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * @author Callum Johnson
+ * @since 11/01/2022 - 23:48
+ */
+public class Riding extends AbstractMovementSkill {
+
+    /**
+     * The Riding skill is levelled through riding entities.
```
Comment (dmccoystephenson) [inline src/main/java/dansplugins/simpleskills/skills/Riding.java:20]: Good idea abstracting this one out.

**Review by dmccoystephenson — APPROVED** [review]
You have massively improved the project. Thank you.

File: src/main/java/dansplugins/simpleskills/commands/SkillCommand.java (hunk 2 above)
Comment (CallumJohnson) [reply src/main/java/dansplugins/simpleskills/commands/SkillCommand.java:38]: Ah, okay, I'll commit a new change for that fix now :) thank you very much for finding one of the last "TODO"s I had left for myself :)

**Pull request merged**